L1Trigger/DTTriggerPhase2/src/MuonPathAssociator.cc

```diff
@@ -612,24 +630,38 @@ void MuonPathAssociator::correlateMPaths(edm::Handle<DTDigiCollection> dtdigis,
                   lat = 0;
                 }
                 if (std::abs(x_inSL1 - x_wire) < minx) {
-                  minx = std::abs(x_inSL1 - x_wire);
-                  next_wire = best_wire;
-                  next_tdc = best_tdc;
-                  next_layer = best_layer;
-                  next_lat = best_lat;
-
+                  // different layer than the stored in best, hit added, matched_digis++;. This approach in somewhat
+                  // buggy, as we could have stored as best LayerX -> LayerY -> LayerX, and this should
+                  // count only as 2 hits. However, as we confirm with at least 2 hits, having 2 or more
+                  // makes no difference
+                  if (dtLId.layer() != best_layer) {
+                    minx = std::abs(x_inSL1 - x_wire);
+                    next_wire = best_wire;
+                    next_tdc = best_tdc;
+                    next_layer = best_layer;
+                    next_lat = best_lat;
+                    matched_digis++;
+                  }
                   best_wire = (*digiIt).wire();
                   best_tdc = (*digiIt).time();
                   best_layer = dtLId.layer();
                   best_lat = lat;
-                  matched_digis++;
                 } else if ((std::abs(x_inSL1 - x_wire) >= minx) && (std::abs(x_inSL1 - x_wire) < min2x)) {
```

Comment:
```suggestion
                } else if (std::abs(x_inSL1 - x_wire) < min2x) {
```

Reply:
> @jaimeleonh , do I understand correctly that the bug fix for #39338 is not included in this PR any more? Or could you please point it out to me which is the code change that fixes it?

The bug fix is included in the changes in L1Trigger/DTTriggerPhase2/src/MPQualityEnhancerFilter.cc.